src/libs/localization-api/localize.controller.ts

```diff
@@ -0,0 +1,152 @@
+
+The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+import {
+	DefaultTranslationSet,
+	FunctionParams,
+	TranslationSet,
+	connectedElements,
+	documentDirection,
+	documentLanguage,
+	fallback,
+	translations,
+} from './manager.js';
+import { UmbController, UmbControllerHost } from '@umbraco-cms/backoffice/controller-api';
+
+const LocalizeControllerAlias = Symbol();
+/**
+ * The UmbLocalizeController enables localization for your element.
+ *
+ * @see UmbLocalizeElement
+ * @example
+ * ```ts
+ * import { UmbLocalizeController } from '@umbraco-cms/backoffice/localization-api';
+ *
+ * \@customElement('my-element')
+ * export class MyElement extends LitElement {
+ *   private localize = new UmbLocalizeController(this);
+ *
+ *   render() {
+ *     return html`<p>${this.localize.term('general_close')}</p>`;
+ *   }
+ * }
+ * ```
+ */
+export class UmbLocalizeController<TranslationType extends TranslationSet = DefaultTranslationSet>
+	implements UmbController
+{
+	#host;
+	#hostEl;
+	controllerAlias = LocalizeControllerAlias;
+
+	constructor(host: UmbControllerHost) {
+		this.#host = host;
+		this.#hostEl = host.getHostElement() as HTMLElement;
+		this.#host.addController(this);
+	}
+
+	hostConnected(): void {
+		if (connectedElements.has(this.#hostEl)) {
+			return;
+		}
+
+		connectedElements.add(this.#hostEl);
+	}
+
+	hostDisconnected(): void {
+		connectedElements.delete(this.#hostEl);
+	}
+
+	destroy(): void {
+		// We do not need to call delete here, as hostDisconnected is called when controller is removed.
+		//connectedElements.delete(this.host);
+		this.#host.removeController(this);
+		this.#hostEl = undefined as any;
+	}
+
+	/**
+	 * Gets the host element's directionality as determined by the `dir` attribute. The return value is transformed to
+	 * lowercase.
+	 */
+	dir() {
+		return `${this.#hostEl.dir || documentDirection}`.toLowerCase();
+	}
+
+	/**
+	 * Gets the host element's language as determined by the `lang` attribute. The return value is transformed to
+	 * lowercase.
+	 */
+	lang() {
+		return `${this.#hostEl.lang || documentLanguage}`.toLowerCase();
+	}
+
+	private getTranslationData(lang: string) {
```

Comment:
This method is only being used once, and there we only use two of the returned props. I think we just just get rid of this and copy the code for retrieving primary and secondary into the method below.